biomage/experiment/download.py

```diff
@@ -0,0 +1,338 @@
+        end_message = "Raw RDS files have been downloaded."
+    else:
+        file_name = "processed_r.rds"
+        bucket = f"{PROCESSED_RDS_BUCKET}-{input_env}"
+        end_message = "Processed RDS files have been downloaded."
+
+    key = f"{experiment_id}/r.rds"
+    file_path = output_path / file_name
+
+    _download_file(bucket, key, file_path)
+
+    print(f"RDS file saved to {file_path}")
+    print(end_message)
+
+
+def _download_cellsets(experiment_id, input_env, output_path):
+    FILE_NAME = "cellsets.json"
+
+    bucket = f"{CELLSETS_BUCKET}-{input_env}"
+    key = experiment_id
+    file_path = output_path / FILE_NAME
+    _download_file(bucket, key, file_path)
+    print(f"Cellsets file saved to {file_path}")
+    print("Cellsets file have been downloaded.")
+
+
+@click.command()
+@click.option(
+    "-e",
+    "--experiment_id",
+    required=True,
+    help="Experiment ID to be copied.",
+)
+@click.option(
+    "-i",
+    "--input_env",
+    required=True,
+    help="Input environment to pull the data from.",
+)
+@click.option(
+    "-o",
+    "--output_path",
+    required=False,
+    default=None,
+    help="Name of output folder. By default this will be the experiment id.",
+)
+@click.option(
+    "-a",
+    "--all",
+    required=False,
+    is_flag=True,
+    default=False,
+    help="Download all files for the experiment.",
+)
+@click.option(
+    "--name_with_id",
+    required=False,
+    is_flag=True,
+    default=False,
+    help="Use sample id to name samples.",
+)
+@click.option(
+    "-f",
+    "--files",
+    multiple=True,
+    required=False,
+    default=[SAMPLES],
+    help=(
+        "Files to download. By default only the samples (-f samples) are downloaded. "
+        "You can also download cellsets (-f cellsets), raw RDS (-f raw_rds) and "
+        "processed RDS (-f qc_rds)."
+    ),
+)
+def download(experiment_id, input_env, output_path, files, all, name_with_id):
+    """
+    Downloads files associated with an experiment from a given environment.\n
+
+    E.g.:
+    biomage experiment download -i staging -e 2093e95fd17372fb558b81b9142f230e
+    -f samples -f cellsets -o output/folder
+    """
+
+    # Set output path
+    # By default, the output path is named after the experiment id
+    if not output_path:
+        output_path = experiment_id
+
+    output_path = Path(os.getcwd()) / output_path
+
+    print("Saving downloaded files to: ", str(output_path))
+
+    selected_files = []
+    if all:
+        selected_files = [SAMPLES, CELLSETS, RAW_RDS, PROCESSED_RDS]
+    else:
+        selected_files = list(files)
+
+    for file in selected_files:
+        if file == SAMPLES:
+            print("\n== Downloading sample files")
+            try:
+                _download_samples_v2(experiment_id, input_env, output_path, name_with_id)
+            except Exception:
+                _download_samples_v1(experiment_id, input_env, output_path, name_with_id)
```

Comment:
I would remove this and raise an exception instead with the error that occured.

Reply:
Silencing exceptions and doing something majorly different without printing any logs is not a recommended practice, because it can be a pain to debug and if there was some real problem, it makes people realise that there is a problem very further down the line.
You would silence an exception if you are 1000% sure that all possible exceptions are ok. And even then, I would say, you still want to print a log of what happened.
About v1, I don't think that we should support this at all, so I will remove that method alltogether.

Reply:
When I was making the command, I found that not all of the experiments (in staging) have been migrated from DynamoDB to RDS yet, so the command was working for some experiments, but not for others. That's why I decided to create the v1 version, so that the command can be used from now - otherwise we'll have to wait until v2 is up.

Reply:
I was relying on exceptions, but I'll change it to use return values instead.

Reply:
I'll remove support for v1 and tell users to download the files directly from S3 if it's not available in v2.

Reply:
sounds good!